metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -527,7 +527,7 @@ public void replay(RemoveTopicRecord record) {
             if (colliding != null) {
                 colliding.remove(topic.name);
                 if (colliding.isEmpty()) {
-                    topicsWithCollisionChars.remove(topic.name);
+                    topicsWithCollisionChars.remove(normalizedName);
```

Comment:
The key for `topicsWithCollisionChars` should be normalized name, not original name.